### PR TITLE
Summary unit test cases - partial commit

### PR DESCRIPTION
**Pull request by rparupell**
Since code changes are big, creating pull request with reasonable changes to review the code.

### CODE REVIEW

File: internal/summary/summary_test.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+			path: ap,
+		},
+	}
+	fakeSummaryState.Probes[fakeName].Meta["audit_path"] = ap
+	fakeSummaryState.Probes[fakeName].audit.PodsDestroyed = &fakeSummaryState.Probes[fakeName].PodsDestroyed
+	fakeSummaryState.Probes[fakeName].audit.ScenariosAttempted = &fakeSummaryState.Probes[fakeName].ScenariosAttempted
+	fakeSummaryState.Probes[fakeName].audit.ScenariosSucceeded = &fakeSummaryState.Probes[fakeName].ScenariosSucceeded
+	fakeSummaryState.Probes[fakeName].audit.ScenariosFailed = &fakeSummaryState.Probes[fakeName].ScenariosFailed
+	fakeSummaryState.Probes[fakeName].audit.Result = &fakeSummaryState.Probes[fakeName].Result
+	fakeSummaryState.Probes[fakeName].countResults()
```
Comment (eddie-knight) [inline internal/summary/summary_test.go:69]: Please pull this logic out to an independent function

**Review by eddie-knight — CHANGES_REQUESTED** [review]

File: internal/summary/summary_test.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+			actualPosition, actualFound := utils.FindString(loggedPods, tt.args.podName)
+			if !(actualPosition >= 0 && actualFound == true) {
+				t.Errorf("State.Meta doesn't contain pod name: %v", loggedPods)
+			}
+		})
+	}
+}
+
+// createMockProbe - creates a mock summarystate and probe object in it and returns summarySteate object.
+func createMockProbe(probename string) SummaryState {
```
Comment (lcarrazana) [inline internal/summary/summary_test.go:46]: Should the function name be: `createMockSummaryState` ?
Or maybe this: `createSummaryStateWithFakeProbes` ?

What are your thoughts?

Comment (rparupell) [reply internal/summary/summary_test.go]: i changed.committed pls review

File: internal/summary/summary_test.go (hunk 1 above)
Comment (rparupell) [reply internal/summary/summary_test.go]: Done

**Review by eddie-knight — APPROVED** [review]

**Review by lcarrazana — APPROVED** [review]

**Pull request merged**